src/common.jl

```diff
@@ -1,194 +1,258 @@
+    tT .= Taylor1(typeof(t), integrator.alg.order)
+    tT[0] = t
+    uT .= Taylor1.(u, tT.order)
+    duT .= zero.(Taylor1.(u, tT.order))
+    uauxT .= similar(uT)
+    parse_eqs.x = _determine_parsing!(parse_eqs.x, f, tT, uT, duT, p)
+    __jetcoeffs!(Val(parse_eqs.x), f, tT, uT, duT, uauxT, p)
+    # FSAL for interpolation
+    integrator.fsalfirst = fsalfirst
+    integrator.fsallast = k
+    integrator.kshortsize = 1
+    resize!(integrator.k, integrator.kshortsize)
+    integrator.k[1] = integrator.fsalfirst
+    # integrator.f(integrator.fsalfirst,integrator.uprev,integrator.p,integrator.t)
+    integrator.fsalfirst = constant_term.(duT)
+    integrator.destats.nf += 1
+end
 
-    if typeof(prob.u0) <: AbstractArray
-      _timeseries = Vector{uType}(undef, 0)
-      for i=start_idx:size(vectimeseries, 1)
-          push!(_timeseries, reshape(view(vectimeseries, i, :, )', sizeu))
-      end
-    else
-      _timeseries = vec(vectimeseries)[start_idx:end]
+function perform_step!(integrator, cache::TaylorMethodCache)
+    @unpack t, dt, u, f, p = integrator
+    @unpack k, tT, uT, duT, uauxT, parse_eqs = cache
+    evaluate!(uT, dt, u)
+    tT[0] = t+dt
+    for i in eachindex(u)
+        @inbounds uT[i][0] = u[i]
+        duT[i].coeffs .= zero(duT[i][0])
     end
-
-    DiffEqBase.build_solution(prob,  alg, _t, _timeseries,
-        timeseries_errors = timeseries_errors, retcode = :Success)
+    __jetcoeffs!(Val(parse_eqs.x), f, tT, uT, duT, uauxT, p)
+    k = constant_term.(duT) # For the interpolation, needs k at the updated point
+    integrator.destats.nf += 1
 end
 
-# This function is part of OrdinaryDiffEq.jl; MIT-licensed
-# https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/8b5af68ee4126d3772fe690a7128468f1334b4d6/src/solve.jl#L411
-function tstop_saveat_disc_handling(tstops, saveat, d_discontinuities, tspan)
-    t0, tf = tspan
-    tType = eltype(tspan)
-    tdir = sign(tf - t0)
+stepsize_controller!(integrator,alg::_TaylorMethod) = stepsize(integrator.cache.uT, integrator.opts.abstol)
+step_accept_controller!(integrator, alg::_TaylorMethod, q) = q
 
-    tdir_t0 = tdir * t0
-    tdir_tf = tdir * tf
+function DiffEqBase.solve(
+        prob::DiffEqBase.AbstractODEProblem{uType, tupType, isinplace},
+        alg::TaylorMethod, args...;
+        verbose=true,
+        kwargs...) where
+        {uType, tupType, isinplace}
 
-    # time stops
-    tstops_internal = BinaryMinHeap{tType}()
-    if isempty(d_discontinuities) && isempty(tstops) # TODO: Specialize more
-      push!(tstops_internal, tdir_tf)
+    if verbose
+        warned = !isempty(kwargs) && check_keywords(alg, kwargs, warnlist)
+        warned && warn_compat()
+    end
+
+    f = prob.f
+    if !isinplace && typeof(prob.u0) <: AbstractArray
+        if prob.f isa DynamicalODEFunction
+            f1! = (dv, v, u, p, t) -> (dv .= prob.f.f1(v, u, p, t); 0)
+            f2! = (du, v, u, p, t) -> (du .= prob.f.f2(v, u, p, t); 0)
```

Comment:
Since we are in the out-of-place case, shouldn't this _not_ use broadcasting to modify the elements of `dv` in-place?
```suggestion
            f1! = (dv, v, u, p, t) -> (dv = prob.f.f1(v, u, p, t); 0)
            f2! = (du, v, u, p, t) -> (du = prob.f.f2(v, u, p, t); 0)
```